### ISSUE TITLE
`this` is busted while navigating the callstack

### ISSUE THREAD

**Issue by GloriousPtr** [EpicGamesExt/raddebugger#299]
PS: Don't compile with optimization flags
```cpp
struct Test
{
	int x = 0;
	
	void Inc()
	{
		x++;
	}
	void Inc1()
	{
		Inc();
	}
	void Inc2()
	{
		Inc1();
	}
	void Inc3()
	{
		Inc2();
	}
	void Inc4()
	{
		Inc3();
	}
};

int main()
{
	Test t;
	t.Inc4();
}
```

https://github.com/user-attachments/assets/7e9fec2c-5353-4df8-9667-19c73e9bfd62

**Comment by GloriousPtr** [comment]
Here is another case that I can provide source files for: shows different behaviour in Debug and Release builds.
Release: `this` was 0x0 in some frames and for others it was fine
Debug: `this` was 0xcccccc.............. for all frames except the first one
Source files with Debug and Release Builds: https://1drv.ms/u/s!ApjsIRdEL5T4kyGNCSN8r6cvdGLi?e=0cgbMx
exe to debug: `bin/[Debug/Release]/Arc-Editor/Arc-Editor.exe`
Set breakpoint to `ArcGameEngine/Arc/src/Arc/scene/scene.cpp` line: 1151

https://github.com/user-attachments/assets/44da52b4-7572-4d5a-90f2-f1308fc4e1c5

**Comment by ryanfleury** [comment]
Oops, wasn't using correctly unwound registers for the new evaluation... Should be fixed for debug mode in 896f84369b8805f20ea364685fbf5c5adc6dfe17. I think the release mode one is a different issue, investigating now.

**Comment by GloriousPtr** [comment]
Release build seems to be fixed as well
